qa/tasks/deepsea.py

```diff
@@ -0,0 +1,106 @@
+'''
+Task to deploy clusters with DeepSea
+'''
+import logging
+
+from teuthology import misc
+from teuthology.orchestra import run
+from teuthology.salt import Salt
+from teuthology.task import Task
+from util import get_remote_for_role
+
+log = logging.getLogger(__name__)
+
+class DeepSea(Task):
+
+    def __init__(self, ctx, config):
+        super(DeepSea, self).__init__(ctx, config)
+
+        if config is None:
+            config = {}
+        assert isinstance(config, dict), \
+            'deepsea task only accepts a dict for configuration'
+        self.config["repo"] = self.config.get('repo', 'https://github.com/SUSE/DeepSea.git')
+        self.config["branch"] = self.config.get('branch', 'master')
+
+        # determine the role id of the master role
+        if(misc.num_instances_of_type(self.cluster, 'master') != 1):
+            raise ConfigError('deepsea requires a single master role')
+        for id_ in misc.all_roles_of_type(self.ctx.cluster, 'master'):
+            # this loop should iterate only once
+            master_role = '.'.join(['master', str(id_)])
```

Comment:
Why? Because I didn't realize the generator function could be used in that way. Fixing.